### DISCUSSION TITLE
Statistics Preferences: add checkbox for delete warning dialog prefs

### DISCUSSION THREAD

**Discussion by bronwyncombs** [specify/specify7#5849] [Feature Requests]
**Is your feature request related to a problem?**
Some may want the delete warning dialogs and some may be fine with just the cancel feature in edit mode. Also, some may want a warning dialog when deleting other components in the stats page. 

**Describe the solution you'd like**
To the future statistics section in preferences, we should add checkboxes that allow users to toggle having a delete warning dialog.

In addition to the warning dialog for deleting categories that is currently underway ([PR #3722](https://github.com/specify/specify7/pull/3732)), we could also implement warning dialogs for deleting individual statistics and pages. 

As noted in the screenshot, we should discuss what the default settings will be. 

**Additional context**
Slack has this concept in their settings: 

<img width="892" alt="Screenshot 2023-07-20 at 7 44 46 AM" src="https://github.com/specify/specify7/assets/135047322/b651dc2d-f41c-44c9-bf90-b30399cdd5e9">

**Comment by CarolineDenis** [comment]
Decision: no pref available just default behavior
Which is: 
A) Category: Show a warning dialog when deleting a category => TRUE
B) Pages: Show a warning dialog when deleting a page => TRUE
C) Statistics: Show a warning dialog when deleting a statistic => FALSE
@grantfitzsimmons ok for you?
